nixos/modules/hardware/video/nvidia.nix

```diff
@@ -348,15 +356,17 @@ in
       ++ optional (nvidia_x11.persistenced != null && config.virtualisation.docker.enableNvidia)
         "L+ /run/nvidia-docker/extras/bin/nvidia-persistenced - - - - ${nvidia_x11.persistenced}/origBin/nvidia-persistenced";
 
-    boot.extraModulePackages = [ nvidia_x11.bin ];
+    boot.extraModulePackages = if cfg.open then [ nvidia_x11.open ] else [ nvidia_x11.bin ];
```

Comment:
So, just to make sure if I understand:
-  `.out` comes from the runfiel and contains userspace drivers and the `.bin` firmware?
- `.bin` output contains the `.ko` modules?
- `.open` passthru contains the same `.ko` modules but built from source?

Is it explained somewhere in the expression?

Reply:
The proprietary driver does not require the use of a firmware.
`.bin` contains both the modules and some userland tools like `nvidia-smi`.
`.open` is built from a different version of source, the modules in `.bin` is still built from proprietary sources by nvidia.

Reply:
Are any of the tools like nvidia-smi open sourced yet? or are those not included?

Reply:
Here's the gist of `.out` vs `.bin`:
![image](https://user-images.githubusercontent.com/9720532/168239103-8e6f7f02-d8c4-42e1-bb03-ccb44b4aace8.png)

And here's `.open`:
![image](https://user-images.githubusercontent.com/9720532/168239063-57bc3ad8-a3ff-4e8b-843f-6128e0bb6c34.png)

Reply:
I suppose that means we're missing `nvidia-smi` even with nixos module, and we need to fix that

Reply:
`.bin` is always in systemPackages. providing the tools.